Code/GraphMol/Descriptors/EEM.cpp

```diff
@@ -0,0 +1,237 @@
+const float B2[] = {0.0,0.0,   0.0,0.0,0.0,0.0,0.6513,0.6647,0.8410,0.0,   0.0,0.0,0.0,0.0,0.0,0.5759,0.4512,   0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+const float B3[] = {0.0,0.0,   0.0,0.0,0.0,0.0,0.6776,1.4240,0.0,   0.0,   0.0,0.0,0.0,0.0,0.0,0.0,   0.0,      0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+
+/* simple Parameters trained using the small 500 molecules dataset.
+const float kappa = 0.1960;
+
+const float A1[] = {0.0,2.3594,0.0,0.0,0.0,0.0,2.4541,2.5908,2.7130,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.3833};
+const float A2[] = {0.0,0.0,0.0,0.0,0.0,0.0,2.4726,2.5409,2.6766,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.4956};
+const float A3[] = {0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+
+const float B1[] = {0.0,0.5962,0.0,0.0,0.0,0.0,0.2591,0.3316,0.5028,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.4564};
+const float B2[] = {0.0,0.0,0.0,0.0,0.0,0.0,0.2268,0.2319,0.4992,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.1493};
+const float B3[] = {0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0};
+*/
+
+// function to retreive the atomtype value based on the highest bond type of an atom
+// in the publication they don't have access to "Aromatic type"
+double getAtomtype(const RDKit::ROMol mol, const RDKit::Atom *atom) {
+      double t=1.0;
+      double a;
+      RDKit::ROMol::ADJ_ITER nbrIdx, endNbrs;
+      boost::tie(nbrIdx, endNbrs) = mol.getAtomNeighbors(atom);
+      while (nbrIdx != endNbrs) {
+        const RDKit::Bond *bond = mol.getBondBetweenAtoms(*nbrIdx, atom->getIdx());
+        a = bond->getBondTypeAsDouble();
+        if (a == 1.5) {
+             t = 2.0;
+            }
+        if (a == 2.0) {
+             t = 2.0;
+            }
+        if (a == 3.0) {
+             t = 3.0;
+            }
+        ++nbrIdx;
+  }
+  return t;
+}
+
+double* getEEMMatrix(RDKit::ROMol mol, double * dist3D, int n, EEM_arrays EEMatoms) {
+  int sizeArray = (n+1) * (n+1);
+  double* EEM = new double[sizeArray](); // declaration to set all elements to zeros!
+  #define IDX(x, y) (x * (n+1) + y)
+    /* Fill the full n * n block */
+    for(long int i = 0; i < n; i++) {
+        double t = EEMatoms.EEMatomtype[i];
+        int idx = EEMatoms.Atomindex[i];
+        double v;
+        if (t== 1.0) {
+                v=  B1[idx];
+        }
+        if (t== 2.0) {
+                v = B2[idx];
+        }
+        if (t== 3.0) {
+                v = B3[idx];
+        }
+
+        EEM[IDX(i, i)] = v;
+        for(long int j = i + 1; j < n; j++) {
+                EEM[IDX(i, j)] = kappa / dist3D[i*n+j];
+                EEM[IDX(j, i)] = EEM[IDX(i, j)];
+            }
+        }
+    /* Fill last column & row */
+    for(long int i = 0; i < n; i++) {
+        EEM[IDX(n, i)] = 1.0; // column
+        EEM[IDX(i, n)] = -1.0; // row
+
+    }
+    #undef IDX
+    return EEM;
+}
+
+double* getBVector(RDKit::ROMol mol, int n, EEM_arrays EEMatoms) {
```

Comment:
same comment about passing arguments by reference, and shouldn't it be a `const ROMol`?